src/mdio/converters/segy.py

```diff
@@ -494,3 +497,43 @@ def segy_to_mdio(  # noqa: C901
     )
 
     zarr.consolidate_metadata(store_nocache)
+
+
+def _calculate_live_mask_chunksize(grid: Grid) -> Sequence[int] | int:
+    """Calculate the optimal chunksize for the live mask.
+
+    Args:
+        grid: The grid to calculate the chunksize for.
+
+    Returns:
+        Either a single integer (-1) if no chunking is needed, or a sequence of integers
+        representing the optimal chunk size for each dimension of the grid.
+    """
+    if np.sum(grid.live_mask) < INT32_MAX:
+        # Base case where we don't need to chunk the live mask
+        return -1
```

Comment:
If we return the full shape we can always return a Sequence[int], so makes the `| int` unnecessary

```suggestion
        return grid.live_mask.shape
```